packages/dashboard/src/customization/widgets/triangleSymbol/component.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { TriangleWidget } from '../types';
+import { SVG_STROKE_DASHED, SVG_STROKE_DOTTED, SVG_STROKE_SOLID } from '../constants';
+import { useGridSettings } from '~/components/actions/useGridSettings';
+
+const TriangleWidgetComponent: React.FC<TriangleWidget> = (widget) => {
+  const { borderStyle = 'solid', fill = 'none', borderColor = 'black', borderThickness = 5 } = widget.properties;
+  const { cellSize } = useGridSettings();
+
+  let strokeString = SVG_STROKE_SOLID;
+
+  if (borderStyle === 'dashed') {
+    strokeString = SVG_STROKE_DASHED;
+  } else if (borderStyle === 'dotted') {
+    strokeString = SVG_STROKE_DOTTED;
+  }
+
+  const widthPx = widget.width * cellSize;
+  const heightPx = widget.height * cellSize;
+
+  const fitFull = {
+    width: widthPx,
+    height: heightPx,
+  };
+
+  const halfBorderThickness = borderThickness / 2;
+
+  return (
+    <div style={fitFull}>
```

Comment:
looks like we are sharing some common code here with other symbol widgets. What do you think about reorganizing the component structure so that we can make use of the same css classes.

Reply:
Another piece of common code might be width/height in pixels. looks like line anchor uses that calculation too.

Reply:
i think it's a good idea to generalize all shapes (a line is not a shape for this purpose). i will create a directory for shapes within widgets

Reply:
Nice, ya I like that idea, also, it might be useful to have some more utility hooks like useSizeInPixels for instance which would do that calculation of converting grid size to pixels using cell size. I know I saw that code in multiple spots.

Reply:
sounds good. would we want to have that function/hook in the symbol directory or at a higher level? because we might end up using it somewhere else later on too - it's a nice thing to have

Reply:
added it to the symbol library for now but it can be moved anytime